src/mbgl/map/transform_state.hpp

```diff
@@ -20,7 +20,10 @@ class TransformState {
     friend class Transform;
 
 public:
-    TransformState(ConstrainMode = ConstrainMode::HeightOnly, ViewportMode = ViewportMode::Default);
+    constexpr TransformState() = default;
+
+    constexpr TransformState(ConstrainMode constrain_, ViewportMode viewport_)
```

Comment:
This removes the ability to construct with a single argument. (Could you please limit your changes to fixing the main issue, and propose cosmetic changes in a separate PR?)